shared/src/main/scala/squants/QuantityRange.scala

```diff
@@ -21,7 +21,9 @@ import scala.annotation.tailrec
  * @tparam A the Quantity Type
  */
 case class QuantityRange[A <: Quantity[A]](lower: A, upper: A) {
-  if (lower >= upper) throw new IllegalArgumentException("QuantityRange upper bound must be greater than or equal to the lower bound")
+  if (lower >= upper) {
```

Comment:
I wish we were not to throw exceptions on constructors. Instead I'd rather have an `apply` constructor returning an `Option` or `Either` but I guess that would break the API